sdks/java/core/src/main/java/org/apache/beam/sdk/state/Timer.java

```diff
@@ -81,6 +81,9 @@
    */
   void setRelative();
 
+  /** Clears the timer. */
```

Comment:
This contract is vague.

Reply:
There are 2 conditions that we know a cleared timer should never been fired:

- Timer(key = A) is set first then cleared within the same bundle.

- A never-set timer is cleared.

One condition that we know a cleared timer will be fired:

- Timer(key = A) is cleared first then set again.

And the contract is vague under certain conditions(e.g, the runner may or may not clear the timer depends on the implementation details of the runner):

- Timer(key = A)  is being scheduled by service meanwhile user code just clears Timer(key = A). This happens when set and clear calls happen on different bundles.

I'll add ValidatesRunner tests to cover these.